ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -19,6 +19,17 @@ public struct SignalProducer<Value, Error: ErrorType> {
 
 	private let startHandler: (Signal<Value, Error>.Observer, CompositeDisposable) -> ()
 
+	/// Initializes a SignalProducer that will emit the same events as the given signal.
+	///
+	/// If the Disposable returned from start() is disposed or a terminating
+	/// event is sent to the observer, the given signal will be
+	/// disposed.
+	public init<S: SignalType where S.Value == Value, S.Error == Error>(signal: S) {
+		self.init { observer, disposable in
+			disposable += signal.observe(observer)
+		}
+	}
```

Comment:
It seems a little weird to create a `SignalProducer` from a `Signal`. Not necessarily wrong, but a little weird. :confused:

I think you've done this because `merge()`, `switchToLatest()`, and `contact()` are defined on `Signal<SignalProducer<T, E>, E>`. Would it be better to define those on `Signal<Signal<T, E>, E>` instead and implement the `SignalProducer` version with the `Signal` version instead?